pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -158,7 +163,7 @@ func (p *criStatsProvider) ListPodStats() ([]statsapi.PodStats, error) {
 
 		// Fill available stats for full set of required pod stats
 		cs := p.makeContainerStats(stats, container, &rootFsInfo, fsIDtoInfo, podSandbox.GetMetadata().GetUid())
-		p.addPodNetworkStats(ps, podSandboxID, caInfos, cs)
+		p.addPodNetworkStats(ps, podSandboxID, caInfos, cs, containerNetworkStats[podSandboxID])
```

Comment:
is it possible a race condition exists where a new container was created but you don't have stats for it in the dictionary?

Reply:
yep, it's possible. In that case, the network stats is actually nil, so it would just return without doing anything.